src/workers/cdc/primary.ts

```diff
@@ -822,16 +821,11 @@ const setRestoreStreakCache = async (
   streak: ChangeObject<UserStreak>,
 ) => {
   const { userId, currentStreak: previousStreak } = streak;
-  const today = new Date();
-  const shouldAllow = await shouldAllowRestore(con, streak);
-
-  if (!shouldAllow) {
-    return;
-  }
 
   const key = generateStorageKey(StorageTopic.Streak, StorageKey.Reset, userId);
+  const today = new Date();
   const now = today.getTime();
-  const nextDay = addDays(now, 1).setHours(0, 0, 0, 0);
+  const nextDay = addDays(now, 1).setHours(23, 59, 59, 0);
```

Comment:
I think this is not accurate due to:
- I have read on Monday.
- Missed on Tuesday.
- Streak recovery is triggered.
- With this config, it means you can recover until Thursday 23:59 which is not correct anymore (iirc). It should only last until the day the streak was lost.

Reply:
Had similar comment, I think it requires a bit of investigation before removing, but @sshanzel you probably have the best view of everything and how it functions.

Reply:
Yeah, that's why I think it would've helped if we were able to check redis, but now I will try to investigate any existing one, and if we have any data in our redis cache.

Reply:
Right, miss calculated days!